src/main/java/shadows/apotheosis/mixin/ItemStackMixin.java

```diff
@@ -80,4 +92,54 @@ public int swapDura(int amount, int amountCopy, RandomSource pRandom, @Nullable
         }
         return amount - blocked;
     }
+
+    @Unique
+    private static final MutableComponent apotheosis$SPACE = Component.literal(" ");
+
+    private static void appendModifiedEnchTooltip(List<Component> tooltip, Enchantment ench, int realLevel, int nbtLevel) {
+        MutableComponent mc = ench.getFullname(realLevel).copy();
+        mc.getSiblings().clear();
+        Component nbtLevelComp = Component.translatable("enchantment.level." + nbtLevel);
+        Component realLevelComp = Component.translatable("enchantment.level." + realLevel);
+        if (realLevel != 1 || EnchHooks.getMaxLevel(ench) != 1) mc.append(apotheosis$SPACE).append(realLevelComp);
+
+        int diff = realLevel - nbtLevel;
+        char sign = diff > 0 ? '+' : '-';
+        Component diffComp = Component.translatable("(%s " + sign + " %s)", nbtLevelComp, Component.translatable("enchantment.level." + Math.abs(diff))).withStyle(ChatFormatting.DARK_GRAY);
+        mc.append(apotheosis$SPACE).append(diffComp);
+        if (realLevel == 0) {
+            mc.withStyle(ChatFormatting.DARK_GRAY);
+        }
+        tooltip.add(mc);
+    }
+
+    /**
+     * Rewrites the enchantment tooltip lines to include the effective level, as well as the (NBT + bonus) calculation.
+     */
+    @SuppressWarnings("deprecation")
+    @Redirect(method = "getTooltipLines", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/item/ItemStack;appendEnchantmentNames(Ljava/util/List;Lnet/minecraft/nbt/ListTag;)V"))
+    public void apoth_enchTooltipRewrite(List<Component> tooltip, ListTag tagEnchants) {
```

Comment:
Needs to await a solution for https://github.com/Shadows-of-Fire/Apotheosis/issues/1165